### PR TITLE
add testcase for plugin properties should not be revised by compile_m…

### PR DESCRIPTION
**Pull request by wgzintel**
…odel

### Details:
 - *add testcase for plugin properties should not be revised by compile_model*

### Tickets:
 - *ticket-85844*

### CODE REVIEW

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 1]
```diff
@@ -96,6 +112,26 @@ TEST_P(OVPropertiesDefaultTests, CheckDefaultValues) {
...
+TEST_P(OVSetPropComplieModleGetPropTests, SetPropertyComplieModelGetProperty) {
+    OV_ASSERT_NO_THROW(core->set_property(device_name, properties));
+
+    ov::CompiledModel exeNetWork;
+    OV_ASSERT_NO_THROW(exeNetWork = core->compile_model(model, device_name, compileModelProperties));
+
+    for (const auto& property_item : compileModelProperties) {
+        Any exeNetProperty;
+        OV_ASSERT_NO_THROW(exeNetProperty = exeNetWork.get_property(property_item.first));
+        ASSERT_EQ(property_item.second.as<std::string>(), exeNetProperty.as<std::string>());
```
Comment (yangwang201911) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:124]: what about checking property correctness directly following below code?
`ASSERT_EQ(property_item.second, exeNetProperty);`

File: src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp [hunk 2]
```diff
@@ -36,4 +36,45 @@ INSTANTIATE_TEST_SUITE_P(smoke_AutoMultiBehaviorTests, OVPropertiesTests,
...
+INSTANTIATE_TEST_SUITE_P(smoke_cpuCompileModelBehaviorTests,
+                         OVSetPropComplieModleGetPropTests,
+                         ::testing::Combine(::testing::Values(CommonTestUtils::DEVICE_CPU),
+                                            ::testing::ValuesIn(cpu_plugin_properties),
+                                            ::testing::ValuesIn(cpu_compileModel_properties)),
+                         OVSetPropComplieModleGetPropTests::getTestCaseName);
+
+const std::vector<ov::AnyMap> auto_multi_plugin_properties = {
+    {ov::device::priorities(CommonTestUtils::DEVICE_CPU),
+     {CONFIG_KEY(EXCLUSIVE_ASYNC_REQUESTS), CONFIG_VALUE(YES)},
```
Comment (yangwang201911) [inline src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp:60]: Should we define this property in the properties.hpp like the definition of the other properties? @peterchen-intel 
`static constexpr Property<bool, PropertyMutability::RW> exclusive_async_requests{"EXCLUSIVE_ASYNC_REQUESTS"};`

Comment (peterchen-intel) [reply src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp:60]: Currently, doesn't  plan to have this property in OV 2.0

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp (hunk 1 above)
Comment (wgzintel) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:124]: when checking property ov::hint::allow_auto_batching, ASSERT_EQ(property_item.second, exeNetProperty) will fail. test result: Expected equality of these values:
  value
    Which is: YES
  exeNetProperty
    Which is: YES

File: src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp [hunk 3]
```diff
@@ -36,4 +36,43 @@ INSTANTIATE_TEST_SUITE_P(smoke_AutoMultiBehaviorTests, OVPropertiesTests,
                 ::testing::Values(CommonTestUtils::DEVICE_AUTO, CommonTestUtils::DEVICE_MULTI),
                 ::testing::ValuesIn(multi_Auto_properties)),
         OVPropertiesTests::getTestCaseName);
+
+const std::vector<ov::AnyMap> cpu_plugin_properties = {
+    {ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT),
+     {CONFIG_KEY(EXCLUSIVE_ASYNC_REQUESTS), CONFIG_VALUE(YES)},
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp:42]: Remove IE configuration (API 1.0) only key/value

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 4]
```diff
@@ -44,6 +44,22 @@ void OVPropertiesTests::TearDown() {
     }
 }
 
+std::string OVSetPropComplieModleGetPropTests::getTestCaseName(testing::TestParamInfo<CompileModelPropertiesParams> obj) {
+    std::string device_name;
+    AnyMap properties;
+    AnyMap compileModelProperties;
+    std::tie(device_name, properties, compileModelProperties) = obj.param;
+    std::ostringstream result;
+    result << "device_name=" << device_name << "_setProperties_compileModelProperties_getProperties";
```
Comment (songbell) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:53]: the case name is not unique here, better to combine with input parameters

File: src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp (hunk 3 above)
Comment (wgzintel) [reply src/tests/functional/plugin/cpu/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp]: done

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp (hunk 4 above)
Comment (wgzintel) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp]: done

**Review by peterchen-intel — APPROVED** [review]

**Pull request merged**